### DISCUSSION TITLE
acme.sh installation without access to github

### DISCUSSION THREAD

**Discussion by Obyka** [acmesh-official/acme.sh#4772] [Ideas]
It would be nice if it was possible to install acme.sh on a server that can't access https://api.github.com/ or https://api.github.com/.
In such a scenario, a user could simple send acme.sh directory to their server and launch the installation as usual.
Currently, I identified two occurrences where access to github is necessary:

- During the cloning process
- After the installation per se, when acme.sh tries to retrieve the last commit hash with the API.

I imagine that it would be impossible to upgrade, but it could be good enough in certain cases.

A [installation from the archive](https://github.com/acmesh-official/acme.sh/discussions/4671) does not fully answer the issue since the UPGRADING_HASH is still retrieved.

**Comment by neilpang** [comment]
you can just install for the first time,  and then just  copy and overwrite the file "acme.sh" with newer version.
you don't need to "install" everytime.

**Reply by Obyka** [reply]
Thanks for you answer. I understand that you only need to install once but I'm not sure it addresses the issue.

During the installation process, acme.sh tries to get the last commit hash and raises an error if github is not accessible. 
Does it prevent the installation process to finish ?  I'm talking about the line #6560 https://github.com/acmesh-official/acme.sh/pull/4302/commits/0a47f48191064680b6fce97b0d05f8805f4a8727

Would it be better to add a flag that prevents the installation process from retrieving the last commit on GH ?
